### ISSUE TITLE
"(http_protocol) index events with fieldref in routing value behaves like a routing indexer" test fails with 2.0.0-rc1

### ISSUE THREAD

**Issue by untergeek** [logstash-plugins/logstash-output-elasticsearch#268]
```
(http_protocol) index events with fieldref in routing value
  behaves like a routing indexer
Failed (Expected 10498, but got 0). Sleeping for 0.02
Failed (Expected 10498, but got 0). Sleeping for 0.04
Failed (Expected 10498, but got 0). Sleeping for 0.08
Failed (Expected 10498, but got 0). Sleeping for 0.16
Failed (Expected 10498, but got 0). Sleeping for 0.32
Failed (Expected 10498, but got 0). Sleeping for 0.64
Failed (Expected 10498, but got 0). Sleeping for 1.28
Failed (Expected 10498, but got 0). Sleeping for 2.0
Failed (Expected 10498, but got 0). Sleeping for 2.0
Failed (Expected 10498, but got 0). Sleeping for 2.0
    ships events (FAILED - 1)
```

```
Failures:

  1) (http_protocol) index events with fieldref in routing value behaves like a routing indexer ships events
     Failure/Error: insist { cur_count } == event_count
     Insist::Failure:
       Expected 10498, but got 0
     Shared Example Group: "a routing indexer" called from ./spec/integration/outputs/routing_spec.rb:54
     # /Users/buh/.rvm/gems/jruby-1.7.19/gems/insist-1.0.0/lib/insist/assert.rb:8:in `assert'
     # /Users/buh/.rvm/gems/jruby-1.7.19/gems/insist-1.0.0/lib/insist/comparators.rb:14:in `=='
     # ./spec/integration/outputs/routing_spec.rb:33:in `(root)'
     # /Users/buh/.rvm/gems/jruby-1.7.19/gems/stud-0.0.22/lib/stud/try.rb:79:in `try'
     # /Users/buh/.rvm/gems/jruby-1.7.19/gems/stud-0.0.22/lib/stud/try.rb:95:in `try'
     # /Users/buh/.rvm/gems/jruby-1.7.19/gems/stud-0.0.22/lib/stud/try.rb:91:in `try'
     # /Users/buh/.rvm/gems/jruby-1.7.19/gems/stud-0.0.22/lib/stud/try.rb:123:in `try'
     # ./spec/integration/outputs/routing_spec.rb:27:in `(root)'
     # /Users/buh/.rvm/gems/jruby-1.7.19/gems/rspec-wait-0.0.7/lib/rspec/wait.rb:46:in `(root)'
```

**Comment by andrewvc** [comment]
Hmm, I currently have no failing specs on my box with RC1.  Any ideas here @untergeek ?

**Comment by untergeek** [comment]
I'll give it another run with the current release of the plugin and see what shakes out...

But not until Monday :smile:

**Comment by untergeek** [comment]
@andrewvc I got a different failure this time with 2.0:

```

Failures:

  1) failures in bulk class expected behavior should retry actions with response status of 503
     Failure/Error: Unable to find matching line from backtrace
       (LogStash::Outputs::ElasticSearch: {"manage_template"=>true, "index"=>"logstash-2014.11.17", "template_overwrite"=>true, "hosts"=>["127.0.0.1"], "port"=>9200, "retry_max_items"=>10, "retry_max_interval"=>1, "max_retries"=>3, "type"=>"", "tags"=>[], "exclude_tags"=>[], "codec"=><LogStash::Codecs::Plain charset=>"UTF-8">, "workers"=>1, "template_name"=>"logstash", "flush_size"=>500, "idle_flush_time"=>1, "action"=>"index", "path"=>"/", "ssl"=>false, "ssl_certificate_verification"=>true, "sniffing"=>false, "sniffing_delay"=>5, "doc_as_upsert"=>false, "upsert"=>""}).submit([["index", {:_id=>nil, :_index=>"logstash-2014.11.17", :_type=>"logs", :_routing=>nil}, #<LogStash::Event:0x7e8e9777 @metadata_accessors=#<LogStash::Util::Accessors:0x6c40a151 @store={"retry_count"=>3}, @lut={}>, @cancelled=false, @data={"somevalue"=>100, "@timestamp"=>"2014-11-17T20:37:17.223Z", "@version"=>"1"}, @metadata={"retry_count"=>3}, @accessors=#<LogStash::Util::Accessors:0x1ea16838 @store={"somevalue"=>100, "@timestamp"=>"2014-11-17T20:37:17.223Z", "@version"=>"1"}, @lut={"type"=>[{"somevalue"=>100, "@timestamp"=>"2014-11-17T20:37:17.223Z", "@version"=>"1"}, "type"]}>>]])
           expected: 3 times with arguments: ([["index", {:_id=>nil, :_routing=>nil, :_index=>"logstash-2014.11.17", :_type=>"logs"}, #<LogStash::Event:0x7e8e9777 @metadata_accessors=#<LogStash::Util::Accessors:0x6c40a151 @store={"retry_count"=>3}, @lut={}>, @cancelled=false, @data={"somevalue"=>100, "@timestamp"=>"2014-11-17T20:37:17.223Z", "@version"=>"1"}, @metadata={"retry_count"=>3}, @accessors=#<LogStash::Util::Accessors:0x1ea16838 @store={"somevalue"=>100, "@timestamp"=>"2014-11-17T20:37:17.223Z", "@version"=>"1"}, @lut={"type"=>[{"somevalue"=>100, "@timestamp"=>"2014-11-17T20:37:17.223Z", "@version"=>"1"}, "type"]}>>]])
           received: 4 times with arguments: ([["index", {:_id=>nil, :_index=>"logstash-2014.11.17", :_type=>"logs", :_routing=>nil}, #<LogStash::Event:0x7e8e9777 @metadata_accessors=#<LogStash::Util::Accessors:0x6c40a151 @store={"retry_count"=>3}, @lut={}>, @cancelled=false, @data={"somevalue"=>100, "@timestamp"=>"2014-11-17T20:37:17.223Z", "@version"=>"1"}, @metadata={"retry_count"=>3}, @accessors=#<LogStash::Util::Accessors:0x1ea16838 @store={"somevalue"=>100, "@timestamp"=>"2014-11-17T20:37:17.223Z", "@version"=>"1"}, @lut={"type"=>[{"somevalue"=>100, "@timestamp"=>"2014-11-17T20:37:17.223Z", "@version"=>"1"}, "type"]}>>]])
     # ./lib/logstash/outputs/elasticsearch.rb:514:in `retry_flush'
     # ./lib/logstash/outputs/elasticsearch.rb:294:in `register'

Finished in 57.03 seconds (files took 4.66 seconds to load)
25 examples, 1 failure

Failed examples:

rspec ./spec/integration/outputs/retry_spec.rb:77 # failures in bulk class expected behavior should retry actions with response status of 503
```

**Comment by untergeek** [comment]
This has been resolved in #281
